app.py

```
@@ -34,25 +41,200 @@
     # Add more semesters as needed
 }
 
+# --- Password Hashing Helpers ---
+def hash_password(password):
```

Ruff (0.8.2)

Missing return type annotation for public function `hash_password`

(ANN201)